pycfmodel/model/intrinsic_function_resolver.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright 2018-2019 Skyscanner Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+"""
+import logging
+
+from base64 import b64encode
+from typing import Dict
+
+from pycfmodel.model.regexs import CONTAINS_CF_PARAM
+
+
+class IntrinsicFunctionResolver:
+    def __init__(self, params: Dict, mappings: Dict):
+        self.params = params
+        self.mappings = mappings
+        self.functions = {
+            "Ref": self._ref,
+            "Fn::ImportValue": self._ref,
+            "Fn::Join": self._join,
+            "Fn::FindInMap": self._find_in_map,
+            "Fn::Sub": self._sub,
+            "Fn::Select": self._select,
+            "Fn::Split": self._split,
+            "Fn::If": self._if,
+            "Fn::And": self._and,
+            "Fn::Or": self._or,
+            "Fn::Not": self._not,
+            "Fn::Equals": self._equals,
+            "Fn::Base64": self._base64,
+            "Fn::GetAtt": self._get_attr,
+            "Condition": self._condition,
+        }
+
+    def default_value(self):
+        logging.warning("Using default value")
+        return "NOVALUE"
+
+    def resolve(self, function):
+        if not isinstance(function, dict):
+            return function
+        (function, function_body), = function.items()
```

Comment:
What happens if there's more than 1 entry in the dict?

Reply:
It would fail with `ValueError: too many values to unpack (expected 1)`

Reply:
Is that something that we should protect against?

Reply:
If there is more than one key, it would mean that the cloudformation is not well formatted. I would wrap it into a custom error saying that CF has errors.